server/rest/src/main/java/org/infinispan/rest/resources/LoggingResource.java

```diff
@@ -65,37 +65,38 @@ public Invocations getInvocations() {
                   .addOptionalParameter("loggerName", loggerName)
                   .addOptionalParameter("level", level)
                   .addOptionalParameter("appenders", appenders)
-      ).handle((o, t) -> {
-         NettyRestResponse.Builder response = new NettyRestResponse.Builder();
-         if (t == null) {
-            response.status(HttpResponseStatus.NO_CONTENT);
+                  .subject(request.getSubject())
+      ).handle((o, t) -> handle(t));
+   }
+
+   private NettyRestResponse handle(Throwable t) {
+      NettyRestResponse.Builder response = new NettyRestResponse.Builder();
+      if (t == null) {
+         response.status(HttpResponseStatus.NO_CONTENT);
+      } else {
+         while (t.getCause() != null) {
+            t = t.getCause();
+         }
+         if (t instanceof IllegalStateException) {
+            response.status(HttpResponseStatus.CONFLICT).entity(t.getMessage());
+         } else if (t instanceof IllegalArgumentException) {
+            response.status(HttpResponseStatus.BAD_REQUEST).entity(t.getMessage());
+         } else if (t instanceof NoSuchElementException) {
+            response.status(HttpResponseStatus.NOT_FOUND).entity(t.getMessage());
+         } else if (t instanceof SecurityException) {
+            response.status(HttpResponseStatus.FORBIDDEN).entity(t.getMessage());
          } else {
-            while (t.getCause() != null) {
-               t = t.getCause();
-            }
-            if (t instanceof IllegalStateException) {
-               response.status(HttpResponseStatus.CONFLICT).entity(t.getMessage());
-            } else if (t instanceof IllegalArgumentException) {
-               response.status(HttpResponseStatus.BAD_REQUEST).entity(t.getMessage());
-            } else if (t instanceof NoSuchElementException) {
-               response.status(HttpResponseStatus.NOT_FOUND).entity(t.getMessage());
-            } else {
-               response.status(HttpResponseStatus.INTERNAL_SERVER_ERROR).entity(t.getMessage());
-            }
+            response.status(HttpResponseStatus.INTERNAL_SERVER_ERROR).entity(t.getMessage());
          }
-         return response.build();
-      });
+      }
+      return response.build();
    }
 
    private CompletionStage<RestResponse> deleteLogger(RestRequest request) {
       TaskManager taskManager = invocationHelper.getServer().getTaskManager();
       String loggerName = request.variables().get("loggerName");
       return taskManager.runTask("@@logging@remove", new TaskContext().addParameter("loggerName", loggerName))
```

Comment:
isn't required the Subject be set in the `TaskContext`?

Reply:
Yep, also added a test for this.